src/limiter/bound_preserving_limiter_factory.cpp

```diff
@@ -0,0 +1,211 @@
+#include "parameters/all_parameters.h"
+#include "bound_preserving_limiter_factory.hpp"
+#include "bound_preserving_limiter.h"
+#include "tvb_limiter.h"
+#include "maximum_principle_limiter.h"
+#include "positivity_preserving_limiter.h"
+
+namespace PHiLiP {
+template <int dim, typename real>
+std::shared_ptr< BoundPreservingLimiter<dim,real> >
```

Comment:
I tried to implement the limiter factory like flow_solver_factory but the place where create_limiter is being called is ode_solver_base which is not templated on nstate. Should I try to template ode_solver_base on nstate to resolve this or is there any other way I can go about implementing this?

Reply:
After using recursive templating on create limiter, you can call `create_limiter()` from ode_solver_base() with nstate set to the value of maximum possible nstate. Recursive templating within create_limiter() would then reduce nstate until it matches the nstate of the problem. For a complete example, you can have a look at how Tests are created. `main.cpp` does not have access to nstate. Yet, it calls `Tests::TestsFactory<dim,nstate>::create_test()`, with `create_test()` in tests.cpp.

Reply:
Thank you for the help! I changed the factory file so that `create_limiter` is recursively templated and then `select_limiter` is used to create the limiter!